R/teal_data-constructor.R

```diff
@@ -19,22 +19,22 @@
 #' @details
 #'
 #' A `teal_data` is meant to be used for reproducibility purposes. The class inherits from
-#' [`teal.data::qenv`] and we encourage to get familiar with \CRANpkg{teal.code} first.
+#' [`teal.code::qenv`] and we encourage to get familiar with \CRANpkg{teal.code} first.
 #' `teal_data` has following characteristics:
 #'
 #' - It inherits from the environment and methods such as [`$`], [get()], [ls()], [as.list()],
 #' [parent.env()] work out of the box.
 #' - `teal_data` is a locked environment, and data modification is only possible through the
-#' [teal.code::eval_code()] and [within.qenv()] functions.
-#' - It stores metadata about the code used to create the data (see [get_code()]).
+#' [teal.code::eval_code()] and [teal.code::within.qenv()] functions.
```

Comment:
@donyunardi 
`within` is a base generic. It is not possible to link method to the package as methods are attached somewhere else. What we can link only is `<method>.<class>`. Note you mentioned in the comment comes from this line.
 
```suggestion
#' [teal.code::eval_code()] and [within.qenv()] functions.
```

Reply:
The NOTE exists prior of attaching the `teal.code::` prefix as mentioned [here] (https://github.com/insightsengineering/teal.data/pull/364#issuecomment-2617349300).

Let's just do `[`within.qenv()`][teal.code::within.qenv()]` then.